src/Plugins/ComplexCore/src/ComplexCore/Filters/CropImageGeometry.cpp

```diff
@@ -0,0 +1,536 @@
+#include "complex/Parameters/MultiArraySelectionParameter.hpp"
+#include "complex/Parameters/NumberParameter.hpp"
+#include "complex/Parameters/StringParameter.hpp"
+#include "complex/Utilities/SamplingUtils.hpp"
+
+namespace complex
+{
+namespace
+{
+template <typename T>
+void copyDataTuple(IDataArray& oldArray, IDataArray& newArray, usize oldIndex, usize newIndex)
+{
+  auto& oldArrayCast = static_cast<DataArray<T>&>(oldArray);
+  auto& newArrayCast = static_cast<DataArray<T>&>(newArray);
+
+  auto numComponents = oldArrayCast.getNumberOfComponents();
+
+  for(usize i = 0; i < numComponents; i++)
+  {
+    usize newOffset = newIndex * numComponents;
+    usize oldOffset = oldIndex * numComponents;
+    newArrayCast[newOffset + i] = oldArrayCast[oldOffset + i];
+  }
+}
+
+void copyDataArrayTuple(IDataArray& oldArray, IDataArray& newArray, usize oldIndex, usize newIndex)
+{
+  auto dataType = oldArray.getDataType();
+  switch(dataType)
+  {
+  case DataType::boolean:
+    copyDataTuple<bool>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::float32:
+    copyDataTuple<float32>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::float64:
+    copyDataTuple<float32>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::int8:
+    copyDataTuple<int8>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::int16:
+    copyDataTuple<int64>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::int32:
+    copyDataTuple<int32>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::int64:
+    copyDataTuple<int64>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::uint8:
+    copyDataTuple<uint8>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::uint16:
+    copyDataTuple<uint64>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::uint32:
+    copyDataTuple<uint32>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  case DataType::uint64:
+    copyDataTuple<uint64>(oldArray, newArray, oldIndex, newIndex);
+    break;
+  default:
+    break;
+  }
+}
+
+IntVec3 getCurrentVolumeDataContainerDimensions(const DataStructure& dataStructure, const DataPath& imageGeomPath)
+{
+  IntVec3 data = {0, 0, 0};
+
+  const auto* image = dataStructure.getDataAs<ImageGeom>(imageGeomPath);
+  if(image != nullptr)
+  {
+    data[0] = image->getNumXPoints();
+    data[1] = image->getNumYPoints();
+    data[2] = image->getNumZPoints();
+  }
+  return data;
+}
+
+FloatVec3 getCurrentVolumeDataContainerResolutions(const DataStructure& dataStructure, const DataPath& imageGeomPath)
+{
+  FloatVec3 data = {0, 0, 0};
+  const auto* image = dataStructure.getDataAs<ImageGeom>(imageGeomPath);
+  if(image != nullptr)
+  {
+    data = image->getSpacing();
+  }
+  return data;
+}
+
+std::string getOldBoxDimensions(const DataStructure& dataStructure, const DataPath& imageGeomPath)
```

Comment:
These two functions aren't used right now, but in the original the were used for the preflight updated value.